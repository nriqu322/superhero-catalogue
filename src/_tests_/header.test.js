import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Header from '../components/header';

afterEach(cleanup);

it('renders ok', () => {
  const { asFragment } = render(<Header />);
  expect(asFragment()).toMatchSnapshot();
});

it('check for h1 text', () => {
  const { getByTestId } = render(<Header />);
  expect(getByTestId('header-h1')).toHaveTextContent('Superheroes and Villains');
});

it('check for p text', () => {
  const { getByTestId } = render(<Header />);
  expect(getByTestId('header-p')).toHaveTextContent('search in the catalogue');
});
