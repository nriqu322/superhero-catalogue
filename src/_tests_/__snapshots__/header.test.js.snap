// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ok 1`] = `
<DocumentFragment>
  <div
    class="header-container"
  >
    <header>
      <h1
        data-testid="header-h1"
      >
        Superheroes and Villains
      </h1>
      <p
        data-testid="header-p"
      >
        search in the catalogue
      </p>
    </header>
  </div>
</DocumentFragment>
`;
